File: frontend/src/const.ts
import { OrderStatus } from "@/client";

export const orderStatusStateMachine: Record<OrderStatus, OrderStatus[]> = {
  cancelled: [],
  completed: [],
  on_hold: ["processing"],
  processing: ["completed", "rejected", "on_hold"],
  rejected: [],
  submitted: ["processing", "on_hold", "cancelled"],
};

export const humanReadableStatus: Record<OrderStatus, string> = {
  cancelled: "Cancelled",
  completed: "Completed",
  on_hold: "On hold",
  processing: "Processing",
  rejected: "Rejected",
  submitted: "Submitted",
};

File: backend/backend/const.py
from backend.models.tables import OrderStatus

ORDER_STATUS_STATE_MACHINE: dict[OrderStatus, list[OrderStatus]] = {
    OrderStatus.CANCELLED: [],
    OrderStatus.COMPLETED: [],
    OrderStatus.ON_HOLD: [OrderStatus.PROCESSING],
    OrderStatus.PROCESSING: [OrderStatus.COMPLETED, OrderStatus.REJECTED, OrderStatus.ON_HOLD],
    OrderStatus.REJECTED: [],
    OrderStatus.SUBMITTED: [OrderStatus.PROCESSING, OrderStatus.ON_HOLD, OrderStatus.CANCELLED],
}

WHITELABEL_ORDER_STATUS_MAPPING: dict[OrderStatus, str] = {
    OrderStatus.SUBMITTED: "registered",
    OrderStatus.ON_HOLD: "waiting_for_response",
    OrderStatus.PROCESSING: "in_progress",
    OrderStatus.COMPLETED: "ready",
    OrderStatus.REJECTED: "rejected",
    OrderStatus.CANCELLED: "rejected",
}
